python-package/xgboost/sklearn.py

```diff
@@ -679,3 +679,158 @@ class XGBRegressor(XGBModel, XGBRegressorBase):
     # pylint: disable=missing-docstring
     __doc__ = """Implementation of the scikit-learn API for XGBoost regression.
     """ + '\n'.join(XGBModel.__doc__.split('\n')[2:])
+
+
+class XGBRanker(XGBModel):
+    # pylint: disable=missing-docstring,too-many-arguments,invalid-name
+    __doc__ = """Implementation of sklearn API for XGBoost Ranking
+           """ + '\n'.join(XGBModel.__doc__.split('\n')[2:])
+
+    def __init__(self, max_depth=3, learning_rate=0.1, n_estimators=100,
+                 silent=True, objective="rank:pairwise", booster='gbtree',
+                 n_jobs=-1, nthread=None, gamma=0, min_child_weight=1, max_delta_step=0,
+                 subsample=1, colsample_bytree=1, colsample_bylevel=1,
+                 reg_alpha=0, reg_lambda=1, scale_pos_weight=1,
+                 base_score=0.5, random_state=0, seed=None, missing=None, **kwargs):
+
+        super(XGBRanker, self).__init__(max_depth, learning_rate,
+                                        n_estimators, silent, objective, booster,
+                                        n_jobs, nthread, gamma, min_child_weight, max_delta_step,
+                                        subsample, colsample_bytree, colsample_bylevel,
+                                        reg_alpha, reg_lambda, scale_pos_weight,
+                                        base_score, random_state, seed, missing)
+
+    def fit(self, X, y, group, sample_weight=None, eval_set=None, sample_weight_eval_set=None,
+            eval_group=None, eval_metric=None, early_stopping_rounds=None,
+            verbose=False, xgb_model=None):
+        # pylint: disable = attribute-defined-outside-init,arguments-differ
+        """
+        Fit the gradient boosting model
+        Parameters
+        ----------
+        X : array_like
+            Feature matrix with the first feature containing a group indicator
```

Comment:
"with the first feature containing a group indicator". Is this still the case? I thought `group` parameter already contains a group indicator.